frontend/server/src/main/java/com/amazonaws/ml/mms/http/ManagementRequestHandler.java

```diff
@@ -239,16 +239,18 @@ private void updateModelWorkers(
             boolean synchronous,
             final Function<Void, Void> onError) {
         ModelManager modelManager = ModelManager.getInstance();
-        CompletableFuture<Boolean> future =
+        CompletableFuture<ApiStatus> future =
                 modelManager.updateModel(modelName, minWorkers, maxWorkers);
         if (!synchronous) {
             NettyUtils.sendJsonResponse(
-                    ctx, new StatusResponse("Worker updated"), HttpResponseStatus.ACCEPTED);
+                    ctx,
+                    new StatusResponse("Processing Workers scale request..."),
+                    HttpResponseStatus.ACCEPTED);
             return;
         }
         future.thenApply(
                         v -> {
-                            if (!v) {
+                            if (v == ApiStatus.FAIL) {
```

Comment:
We don't really care about ApiStatus returned by future, only thing we need to do is loop through workers, and check current status of each workers.

Even CompletableFuture<Void> should be enough.
As I mentioned, this is the only place need change. All other changes are not necessary.

Reply:
I would like to keep it simple for now.
If we don't have a ideal solution, leave it for future.